backend/src/db/repositories/CommentRepository.ts

```diff
@@ -43,23 +43,36 @@ export default class CommentRepository {
             });
     }
 
-    async getUserComments(userId: number, forUserId: number, page: number, perPage: number): Promise<CommentRawWithUserData[]> {
+    async getUserComments(userId: number, forUserId: number, filter: string, page: number, perPage: number): Promise<CommentRawWithUserData[]> {
         const limitFrom = (page - 1) * perPage;
-        return await this.db.query(`
+        let query = `
                 select c.*, v.vote
                     from comments c
                       left join comment_votes v on (v.comment_id = c.comment_id and v.voter_id = :for_user_id)
                     where
                       c.author_id = :user_id and c.deleted = 0
                     order by c.created_at desc
                     limit :limit_from, :limit_count
-            `,
-            {
-                user_id: userId,
-                for_user_id: forUserId,
-                limit_from: limitFrom,
-                limit_count: perPage
-            });
+            `;
+          const params = {
+              user_id: userId,
+              for_user_id: forUserId,
+              limit_from: limitFrom,
+              limit_count: perPage
+          };
+        if (filter) {
```

Comment:
The queries are mostly identical, should we merge them?

Reply:
done